velox/common/memory/MmapAllocator.cpp

```diff
@@ -26,8 +26,14 @@ MmapAllocator::MmapAllocator(const Options& options)
     : kind_(MemoryAllocator::Kind::kMmap),
       useMmapArena_(options.useMmapArena),
       capacity_(bits::roundUp(
-          options.capacity / AllocationTraits::kPageSize,
-          64 * sizeClassSizes_.back())) {
+          options.capacity * (100 - options.smallAllocationReservePct) / 100 /
+              AllocationTraits::kPageSize,
+          64 * sizeClassSizes_.back())),
+      mallocCapacityBytes_(
```

Comment:
How about we calculate reservedBytesForMalloc_ first and the delta is capacity_?

Reply:
We don't need any round ups for malloc cap. So any delta can be used directly. This is however not the case for mmap capacity_